lib/src/Base/Stat/CorrelationAnalysis.cxx

```diff
@@ -123,9 +124,29 @@ Point CorrelationAnalysis::PCC(const Sample & inputSample,
 Point CorrelationAnalysis::SRRC(const Sample & inputSample,
                                 const Sample & outputSample)
 {
+  const UnsignedInteger dimension = inputSample.getDimension();
+  if (dimension < 2) throw InvalidDimensionException(HERE) << "Error: input sample must have dimension > 1";
   if (outputSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
   if (inputSample.getSize() != outputSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
-  return SRC(inputSample.rank(), outputSample.rank());
+
+
+
+  // Rank values are in [0,1,2,...,n-1] with n the size
+  LinearLeastSquares regressionAlgorithm(inputSample.rank(), outputSample.rank());
+  regressionAlgorithm.run();
+  // Linear coefficients
+  const Point linear(regressionAlgorithm.getLinear() * Point(1, 1.0));
```

Comment:
@sofianehaddad or:
```
const Point linear(*regressionAlgorithm.getLinear().getImplementation());
```
to avoid using linear algebra to do a type cast (I know it is use in other places in OT...)